platform/msm8916/msm8916-clock.c

```diff
@@ -684,14 +684,150 @@ static struct clk_lookup msm_clocks_8916[] =
 	CLK_LOOKUP("gcc_blsp1_qup5_i2c_apps_clk", gcc_blsp1_qup5_i2c_apps_clk.c),
 };
 
+#define APCS_C0_PLL_BASE				0xb116000
+#define C0_PLL_MODE_REG					0x0
+#define C0_PLL_MODE_PLL_RESET_N				BIT(2)
+#define C0_PLL_MODE_PLL_BYPASSNL			BIT(1)
+#define C0_PLL_MODE_PLL_OUTCTRL				BIT(0)
+#define C0_PLL_L_VAL_REG				0x4
+#define C0_PLL_M_VAL_REG				0x8
+#define C0_PLL_N_VAL_REG				0xC
+#define C0_PLL_USER_CTL_REG				0x10
+#define C0_PLL_USER_CTL_MN_EN				BIT(24)
+#define C0_PLL_USER_CTL_PLLOUT_LV_EARLY			BIT(3)
+#define C0_PLL_USER_CTL_PLLOUT_LV_BIST			BIT(2)
+#define C0_PLL_USER_CTL_PLLOUT_LV_AUX			BIT(1)
+#define C0_PLL_USER_CTL_PLLOUT_LV_MAIN			BIT(0)
+
+#define C0_PLL_CONFIG_CTL_REG				0x14
+#define C0_PLL_CONFIG_CTL_RESERVE_BIT_30		BIT(30)
+#define C0_PLL_CONFIG_CTL_NMOSC_FREQ_CTRL_PFD_DZSEL_4X	(BIT(27) | BIT(26))
+#define C0_PLL_CONFIG_CTL_CUSEL_250FF			BIT(16)
+#define C0_PLL_CONFIG_CTL_RESERVE_BIT14			BIT(14)
+#define C0_PLL_CONFIG_CTL_CFG_LOCKDET_2PI15		BIT(12)
+#define C0_PLL_CONFIG_CTL_RESERVE_BITS10_0_MAGIC	(BIT(10) | BIT(9) | \
+							 BIT(8)  | BIT(6) | \
+							 BIT(5)  | BIT(2) | BIT(0))
+
+#define C0_PLL_CONFIG_CTL_CFG_RESERVE_BIT30		BIT(30)
+#define C0_PLL_CONFIG_CTL_CFG_RESERVE_BIT14		BIT(14)
+#define C0_PLL_CONFIG_CTL_CFG_LOCKDET_2PI15		BIT(12)
+
+#define APCS_ALIAS0_CMD_RCGR_BASE			0xb111050
+#define APCS_ALIAS0_CMD_RCGR_REG			0x0
+#define APCS_ALIAS0_CMD_RCGR_UPDATE			BIT(0)
+#define APCS_ALIAS0_CFG_REG				0x4
+#define APCS_ALIAS0_CFG_RCGR_SRC_SEL_CLK3		BIT(10)
+#define APCS_ALIAS0_CFG_RCGR_SRC_DIV_2			(BIT(1) | BIT(0))
+#define APCS_ALIAS0_CORE_CBCR_REG			0x8
+#define APCS_ALIAS0_CORE_CBCR_SW_CTL			BIT(0)
+
+static void msm8939_clock_c0_enable_sr2_pll(void)
+{
+	uint32_t base = APCS_C0_PLL_BASE;
+	int regval;
+
+	/* Disable Mode */
+	writel(0, base + C0_PLL_MODE_REG);
+
+	/* Configure L/M/N values - these are accumulator fields */
+	writel(52, base + C0_PLL_L_VAL_REG);
+	writel(0,  base + C0_PLL_M_VAL_REG);
+	writel(1,  base + C0_PLL_N_VAL_REG);
```

Comment:
I'd prefer that configurations like this to have a comment describing values in a more "human friendly" way:

(as per apq8016 TRM, page 2572)
```
/* 19.2 * (52 + 0/1) = 998.4 MHz */
```

Reply:
I've dropped the SR2 config here, we can pick it up in Linux if required.